### PR TITLE
eth/protocols/snap: speed up hash checks

### PR DESCRIPTION
**Pull request by holiman**
This PR optimises the hashing a bit during the snap response verification. We use a `hash` only temporarily, to verify against the request, and can thus reuse the same buffer, and avoid an alloc and copy. We can also use the `keccatState` to avoid a second alloc+copy via `Read` instead of `Sum`.

Benchmark for checking `10K` items with 100 bytes of data each: 
```
BenchmarkHashing/old
BenchmarkHashing/old-6         	     100	  20599259 ns/op	 5120510 B/op	   30002 allocs/op
BenchmarkHashing/new
BenchmarkHashing/new-6         	     219	   5355205 ns/op	     480 B/op	       2 allocs/op
```

### CODE REVIEW

File: eth/protocols/snap/sync.go [hunk 1]
```diff
@@ -2061,14 +2061,14 @@ func (s *Syncer) onByteCodes(peer *Peer, id uint64, bytecodes [][]byte) error {
 
 	// Cross reference the requested bytecodes with the response to find gaps
 	// that the serving node is missing
-	hasher := sha3.NewLegacyKeccak256()
-
+	hasher := sha3.NewLegacyKeccak256().(crypto.KeccakState)
+	var hash = make([]byte, 32)
```
Comment (karalabe) [inline eth/protocols/snap/sync.go:2065]: `hash := make([]byte, 32)`, Go usually either uses `var` for the default value or `:=` for non defaults. Pls also add a newline after this.

Pls update the others too.

Comment (holiman) [reply eth/protocols/snap/sync.go]: Squashfixed

**Review by karalabe — APPROVED** [review]
SGTM

**Review by DGKSK8LIFE — COMMENTED** [review]
formatting change; lgtm

**Pull request merged**